### ISSUE TITLE
Single Ctrl+C is not enough to cancel `az resource group delete`

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Azure/azure-cli#596]
```
# az resource group delete -n docker-machine
Starting resource group delete
^CTraceback (most recent call last):
  File "/usr/local/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.5/site-packages/azure/cli/__main__.py", line 29, in <module>
    sys.exit(azure.cli.main.main(args))
  File "/usr/local/lib/python3.5/site-packages/azure/cli/main.py", line 55, in main
    cmd_result = APPLICATION.execute(args)
  File "/usr/local/lib/python3.5/site-packages/azure/cli/application.py", line 117, in execute
    result = expanded_arg.func(params)
  File "/usr/local/lib/python3.5/site-packages/azure/cli/commands/__init__.py", line 219, in _execute_command
    return LongRunningOperation('Starting {}'.format(name))(result)
  File "/usr/local/lib/python3.5/site-packages/azure/cli/commands/__init__.py", line 96, in __call__
    self._delay()
  File "/usr/local/lib/python3.5/site-packages/azure/cli/commands/__init__.py", line 89, in _delay
    time.sleep(self.poll_interval_ms / 1000.0)
KeyboardInterrupt
```

the process keeps running after I hit Ctrl+C once... It finally quits when I hit it twice:

```
^CException ignored in: <module 'threading' from '/usr/local/lib/python3.5/threading.py'>
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/threading.py", line 1288, in _shutdown
    t.join()
  File "/usr/local/lib/python3.5/threading.py", line 1054, in join
    self._wait_for_tstate_lock()
  File "/usr/local/lib/python3.5/threading.py", line 1070, in _wait_for_tstate_lock
    elif lock.acquire(block, timeout):
KeyboardInterrupt
```

**Comment by tjprescott** [comment]
@ahmetalpbalkan which shell were you using? When I try this in CMD, it doesn't really matter if I press once or twice, I'm at the mercy of the LRO poller for a while. Is this your experience as well?

**Comment by ahmetalpbalkan** [comment]
@tjprescott I use zsh via oh-my-zsh on OS X. I hit Ctrl+C once, waited like 10 seconds, nothing happened and I went ahead by hitting Ctrl+C again. 

It shouldn't matter where you are in the program. Hitting Ctrl+C once is supposed to exit the program, right?

**Comment by tjprescott** [comment]
@ahmetalpbalkan yes. What is happening is the first Ctrl+C is terminating the CLI, but because you initiated a LRO to delete the resource group, the second terminates that thread. I've confirmed it does the behavior you describe in bash as well. (CMD makes you wait even after the second Ctrl+C).

Out of curiosity, is the reason you are pressing Ctrl+C because you don't want to wait for the resource group delete operation to finish? You just want to "fire and forget"?

**Comment by ahmetalpbalkan** [comment]
@tjprescott 

> What is happening is the first Ctrl+C is terminating the CLI, but because you initiated a LRO to delete the resource group, the second terminates that thread. I've

got it but I don't think any of your users will care about this


> is the reason you are pressing Ctrl+C because you don't want to wait for the resource group delete operation to finish? You just want to "fire and forget"?

exactly. since I mostly do manual management of RGs I just want backend API to continue deleting in the background, so I just keep it around for like 3 seconds. I use `az resource group delete` like this all the time.

**Comment by tjprescott** [comment]
@mayurid moving this to the backlog. PR #757 has been open which would create a workaround solution for the CLI. However https://github.com/Azure/autorest/pull/1379 addresses this at the autorest level but has not been approved yet.

**Comment by mayurid** [comment]
@tjprescott: will work with the Autorest guys to get the fix expedited.

**Comment by derekbekoe** [comment]
Fixed with msrestazure 0.4.3 so closing issue.
see https://github.com/Azure/azure-cli/pull/757#issuecomment-247092882
